tests/data/general-config-definitions/config.yaml

```
@@ -1,3 +1,3 @@
 region:
```

We don't want to clone the same repo for variables, regions, etc.
```suggestion
repository:
  common-defintions:
    git: git@github.com:IAMconsortium/common-definitions.git
    hash:
    release:
region:
    repository:
```

---

> We don't want to clone the same repo for variables, regions, etc.

Do you mean cloning the same repo multiple times? 
If so, this wouldn't happen anyway since the code checks if the repo is present. If it is it is just updated if need be. 

Although I do have to say it might be nice to not have to define the repo url multiple times.

> Plus the "dictionary" of a repo-name to the git-attributes can help to support multiple external repos in the future.

This should already work.

```yaml
region:
  repository: https://github.com/IAMconsortium/common-definitions.git/
  # hash: optionally specify a commit hash of the repository
  # release: optionally specify a release of the repository
variable:
  # use a different repo for the variables
  repository: https://github.com/IAMconsortium/common-definitions.git/
  # hash: optionally specify a commit hash of the repository
  # release: optionally specify a release of the repository
```
unless you mean multiple external repos for a single dimension. In this case the code would need to be modified but I'd say we cross that bridge when we get there.